Adyen/UI/Form/Items/Address/FormFullAddressItem.swift

```diff
@@ -64,76 +62,98 @@ public final class FullFormAddressItem: FormValueItem<AddressInfo, AddressStyle>
-        houseNumberTextItem.validationFailureMessage = validationMessage
-        houseNumberTextItem.identifier = ViewIdentifierBuilder.build(scopeInstance: self, postfix: "houseNumber")
+        item.identifier = ViewIdentifierBuilder.build(scopeInstance: self, postfix: field.rawValue)
+        return item
+    }
 
-        return houseNumberTextItem
-    }()
+    private func createPickerItem(from viewModel: AddressViewModel, subRegions: [Region]) -> FormItem {
+        let defaultRegion = subRegions.first { $0.identifier == initialCountry } ?? subRegions[0]
+        let item = FormRegionPickerItem(preselectedValue: defaultRegion,
+                                        selectableValues: subRegions,
+                                        style: style.textField)
+        item.title = localizedString(viewModel.labels[.stateOrProvince], localizationParameters)
 
-    internal lazy var apartmentTextItem: FormTextInputItem = {
-        let apartmentTextItem = FormTextInputItem(style: style.textField)
-        let optionalText = localizedString(.fieldTitleOptional, localizationParameters)
-        let titleText = localizedString(.apartmentSuiteFieldTitle, localizationParameters)
-        let placeholderText = localizedString(.apartmentSuiteFieldPlaceholder, localizationParameters)
-        apartmentTextItem.title = "\(titleText) \(optionalText)"
-        apartmentTextItem.validator = LengthValidator(minimumLength: 0, maximumLength: 70)
-        apartmentTextItem.placeholder = "\(placeholderText) \(optionalText)"
-        apartmentTextItem.identifier = ViewIdentifierBuilder.build(scopeInstance: self, postfix: "apartmentSuite")
-
-        return apartmentTextItem
-    }()
+        bind(item: item, to: .stateOrProvince)
+        return item
+    }
 
-    internal lazy var cityTextItem: FormTextInputItem = {
-        let cityTextItem = FormTextInputItem(style: style.textField)
-        cityTextItem.title = localizedString(.cityFieldTitle, localizationParameters)
-        cityTextItem.placeholder = localizedString(.cityFieldPlaceholder, localizationParameters)
-        cityTextItem.validator = LengthValidator(minimumLength: 2, maximumLength: 70)
-        cityTextItem.validationFailureMessage = validationMessage
-        cityTextItem.identifier = ViewIdentifierBuilder.build(scopeInstance: self, postfix: "city")
+    private func createTextItem(for field: AddressFields, from viewModel: AddressViewModel) -> FormItem {
+        let item = FormTextInputItem(style: style.textField)
+        item.validationFailureMessage = validationMessage
+        item.title = localizedString(viewModel.labels[field], localizationParameters)
+        item.placeholder = localizedString(viewModel.placeholder[field], localizationParameters)
 
-        return cityTextItem
-    }()
+        if viewModel.optionalFields.contains(field) {
+            item.title.map { item.title = $0 + " \(optionalMessage)" }
```

Comment:
that's weird, you're setting `item.title` inside the map closure? is that what you mean?
```
item.title = item.title.map { $0 + " \(optionalMessage)" }
```

or maybe
```
item.title.append(" \(optionalMessage)")
```
?

Reply:
here `map` is like `if let ...`. I only want to append if `item.title` is not nil.

Reply:
i'm a big fan of using `map` everywhere, but here might be not the best case. the `transform` closure ideally should be w/o side-effects. i think in this case `if let` suits better.

Reply:
not a big fun of embedded IFs, but ok..

Reply:
does it have to be an embedded if?
maybe something like
```
        if viewModel.optionalFields.contains(field),
           let itemTitle = item.title {
            item.title = [itemTitle, optionalMessage].joined(separator: " ")
        } else {
            item.validator = LengthValidator(minimumLength: 2, maximumLength: 70)
        }
```